internal/activities/validate_file_formats.go

```diff
@@ -2,75 +2,65 @@ package activities
 
 import (
 	"context"
+	"encoding/csv"
 	"fmt"
+	"io"
 	"io/fs"
+	"os"
 	"path/filepath"
 
+	"go.artefactual.dev/tools/temporal"
+
 	"github.com/artefactual-sdps/preprocessing-sfa/internal/fformat"
 	"github.com/artefactual-sdps/preprocessing-sfa/internal/sip"
 )
 
 const ValidateFileFormatsName = "validate-file-formats"
 
-type ValidateFileFormatsParams struct {
-	SIP sip.SIP
-}
-
-type ValidateFileFormatsResult struct {
-	Failures []string
-}
+type (
+	ValidateFileFormats struct {
+		cfg fformat.Config
+	}
+	ValidateFileFormatsParams struct {
+		SIP sip.SIP
+	}
+	ValidateFileFormatsResult struct {
+		Failures []string
+	}
+)
 
-type ValidateFileFormats struct{}
+type formatList map[string]struct{}
 
-func NewValidateFileFormats() *ValidateFileFormats {
-	return &ValidateFileFormats{}
+func NewValidateFileFormats(cfg fformat.Config) *ValidateFileFormats {
+	return &ValidateFileFormats{cfg: cfg}
 }
 
 func (a *ValidateFileFormats) Execute(
 	ctx context.Context,
 	params *ValidateFileFormatsParams,
 ) (*ValidateFileFormatsResult, error) {
 	var failures []string
+	logger := temporal.GetLogger(ctx)
 
-	sf := fformat.NewSiegfriedEmbed()
-	// TODO(daniel): make allowed list configurable.
-	allowed := map[string]struct{}{
-		"fmt/95":    {},
-		"x-fmt/16":  {},
-		"x-fmt/21":  {},
-		"x-fmt/22":  {},
-		"x-fmt/62":  {},
-		"x-fmt/111": {},
-		"x-fmt/282": {},
-		"x-fmt/283": {},
-		"fmt/354":   {},
-		"fmt/476":   {},
-		"fmt/477":   {},
-		"fmt/478":   {},
-		"x-fmt/18":  {},
-		"fmt/161":   {},
-		"fmt/1196":  {},
-		"fmt/1777":  {},
-		"fmt/353":   {},
-		"x-fmt/392": {},
-		"fmt/1":     {},
-		"fmt/2":     {},
-		"fmt/6":     {},
-		"fmt/141":   {},
-		"fmt/569":   {},
-		"fmt/199":   {},
-		"fmt/101":   {},
-		"fmt/142":   {},
-		"x-fmt/280": {},
-		"fmt/1014":  {},
-		"fmt/1012":  {},
-		"fmt/654":   {},
-		"fmt/1013":  {},
-		"fmt/1011":  {},
-		"fmt/653":   {},
+	if a.cfg.AllowlistPath == "" {
+		logger.Info("ValidateFileFormats: No file format allowlist path set, skipping file format validation")
+
+		return nil, nil
+	}
+
+	f, err := os.Open(a.cfg.AllowlistPath)
+	if err != nil {
+		return nil, fmt.Errorf("ValidateFileFormats: %v", err)
```

Comment:
The `%v` vs. `%w` situation always makes me wonder. I think we were trying to use `%w` in most cases to be able to unwrap it if it's needed.

Reply:
@sevein is a very strong proponent of `%v` in almost all cases.  My understanding is that `%w` shouldn't be used when you don't want to expose the underlying implementation of the error.  I don't see any reason we'd want to unwrap the `os.Open` error here.